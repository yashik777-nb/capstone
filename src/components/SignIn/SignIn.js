import React from "react";
import Avatar from "@material-ui/core/Avatar";
import Button from "@material-ui/core/Button";
import CssBaseline from "@material-ui/core/CssBaseline";
import TextField from "@material-ui/core/TextField";
import { blue } from "@material-ui/core/colors";
import Link from "@material-ui/core/Link";
import Grid from "@material-ui/core/Grid";
import Box from "@material-ui/core/Box";
import LockOutlinedIcon from "@material-ui/icons/LockOutlined";
import Typography from "@material-ui/core/Typography";
import Container from "@material-ui/core/Container";
import { withStyles } from "@material-ui/core/styles";

import CapstoneAPI from "../../data/CapstoneAPI";

const styles = (theme) => ({
  paper: {
    marginTop: 8,
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
  },
  avatar: {
    margin: 10,
    backgroundColor: blue[400],
  },
  form: {
    width: "100%", // Fix IE 11 issue.
    marginTop: 1,
  },
  submit: {
    marginTop: 5,
    marginBottom: 0,
    marginLeft: 2,
  },
  register: {
    marginTop: 5,
  },
});

class SignInMUI extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      username: "",
      password: "",
      error: false,
      errorText: "",
    };
  }

  toRegister(e) {
    e.preventDefault();
    this.props.register();
  }

  onSubmit(e) {
    e.preventDefault();
    const userFound = CapstoneAPI.checkUser(this.state.username);
    if (userFound.length === 1) {
      this.props.onSave(userFound);
    } else {
      this.setState({
        ...this.state,
        errorText: "Username does not exist in database",
      });
    }
  }

  usernameChangeHandler(e) {
    this.setState({
      username: e.target.value,
    });
  }

  passwordChangeHandler(e) {
    this.setState({
      password: e.target.value,
    });
  }

  render() {
    const { classes } = this.props;
    return (
      <Container component="main" maxWidth="xs">
        <CssBaseline />
        <div className={classes.paper}>
          <Avatar className={classes.avatar}>
            <LockOutlinedIcon />
          </Avatar>
          <Typography component="h1" variant="h5">
            Sign in
          </Typography>
          <form className={classes.form} onSubmit={(e) => this.onSubmit(e)}>
            <TextField
              variant="outlined"
              margin="normal"
              required
              fullWidth
              id="email"
              label="Email Address"
              name="email"
              autoComplete="email"
              autoFocus
              value={this.state.username}
              onChange={(e) => this.usernameChangeHandler(e)}
              color="primary"
              error={this.state.errorText.length === 0 ? false : true}
              helperText={this.state.errorText}
            />
            <TextField
              variant="outlined"
              margin="normal"
              required
              fullWidth
              name="password"
              label="Password"
              type="password"
              id="password"
              autoComplete="current-password"
              value={this.state.password}
              onChange={(e) => this.passwordChangeHandler(e)}
            />
            <Button
              type="submit"
              fullWidth
              variant="contained"
              color="primary"
              className={classes.submit}
            >
              Sign In
            </Button>
            <Grid container className={classes.register}>
              <Grid item>
                <Link
                  href="#"
                  variant="body2"
                  onClick={(e) => this.toRegister(e)}
                >
                  {"Don't have an account? Register"}
                </Link>
              </Grid>
            </Grid>
          </form>
        </div>
        <Box mt={8}>
          <Typography variant="body2" color="textSecondary" align="center">
            {"Copyright Â© "}
            <Link color="inherit" href="https://material-ui.com/">
              My Work
            </Link>{" "}
            {new Date().getFullYear()}
            {"."}
          </Typography>
        </Box>
      </Container>
    );
  }
}

export default withStyles(styles, { withTheme: true })(SignInMUI);
