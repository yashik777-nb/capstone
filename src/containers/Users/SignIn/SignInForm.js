import React from "react";
import { withRouter } from "react-router";
import { connect } from "react-redux";
import SignIn from "../../../components/SignIn/SignIn";
import * as actionCreators from "../../../store/actions/actionsIndex";

class SingInForm extends React.Component {
  onSave(user) {
    this.props.authenticateUser(user);
    this.props.history.push("/issues");
  }

  onRegister() {
    this.props.history.push("/register");
  }

  render() {
    return (
      <SignIn
        onSave={(user) => this.onSave(user)}
        register={() => this.onRegister()}
      />
    );
  }
}

// mapDispatchToProps
const mapDispatchToProps = (dispatch) => {
  return {
    authenticateUser: (user) => dispatch(actionCreators.authenticateUser(user)),
  };
};

export default connect(null, mapDispatchToProps)(withRouter(SingInForm));
