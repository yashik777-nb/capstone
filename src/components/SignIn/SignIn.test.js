import React from "react";
import { configure } from "enzyme";
import { BrowserRouter } from "react-router-dom";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { createMount } from "@material-ui/core/test-utils";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import moxios from "moxios";
import toJson from "enzyme-to-json";

import SignIn from "./SignIn";

configure({ adapter: new Adapter() });

const styles = {
  paper: "",
  avatar: "",
  form: "",
  submit: "",
  register: "",
};

const props = {
  classes: styles,
};

const mockFn = jest.fn();

describe("Sign In Component Tests", (done) => {
  let wrapper;

  beforeEach(() => {
    const mount = createMount();
    wrapper = mount(
      <BrowserRouter>
        <SignIn {...props} onSave={mockFn} />
      </BrowserRouter>
    );
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it("Should Have Username Required", () => {
    expect(wrapper.find(TextField).at(0).props().required).toEqual(true);
  });

  it("Update Username to have have On Change method called", () => {
    wrapper
      .find(TextField)
      .at(0)
      .simulate("change", {
        target: { value: "yash@abc.com" },
      });
    wrapper
      .find(TextField)
      .at(1)
      .simulate("change", {
        target: { value: "yash" },
      });
    wrapper.update();
    wrapper.find(Button).simulate("submit");
    moxios.wait(() => {
      let request = moxios.requests.mostRecent();
      wrapper.update();
      expect(JSON.parse(request.config.data)).toMatchObject({
        id: 1,
        username: "yash@abc.com",
        password: "yash",
        firstname: "Yash",
        lastname: "IK",
        location: "UK",
        mobileNumber: 12121212,
      });
      done();
    });
  });

  it("Should Have Password Required", () => {
    expect(wrapper.find(TextField).at(1).props().required).toEqual(true);
  });

  // it("If not Authenticated Should push sign in page", () => {
  //   const button = wrapper.find(Button).at(0);
  //   console.log(button);
  //   const mockFn = jest.fn();
  //   console.log(mockFn);
  //   button.simulate("click");
  //   expect(mockFn).toHaveBeenCalledWith("/signin");
  // });

  test("Should Render Sign In page", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
