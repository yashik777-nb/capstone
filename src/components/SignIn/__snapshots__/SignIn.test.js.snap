// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign In Component Tests Should Render Sign In page 1`] = `ReactWrapper {}`;
